homeassistant/components/sensor/fpl.py

```diff
@@ -0,0 +1,80 @@
+"""
+Sensor for getting daily kwh usage information from Florida Power & Light.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.fpl/
+"""
+from homeassistant.const import ATTR_ATTRIBUTION
+from homeassistant.helpers.entity import Entity
+
+CONF_ATTRIBUTION = "Data provided by fpl.com"
+
+DATA_FPL = 'fpl_data'
+DEPENDENCIES = ['fpl']
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the FPL sensors."""
+    if discovery_info is None:
+        return
+
+    sensors = []
+
+    if 'yesterday_kwh' in discovery_info:
+        sensors.append(FplSensor(
+            hass.data[DATA_FPL], 'yesterday_kwh', 'Yesterday', 'kWh'))
+    if 'yesterday_dollars' in discovery_info:
+        sensors.append(FplSensor(
+            hass.data[DATA_FPL], 'yesterday_dollars', 'Yesterday', 'USD'))
+    if 'mtd_kwh' in discovery_info:
+        sensors.append(FplSensor(
+            hass.data[DATA_FPL], 'mtd_kwh', 'Month to Date', 'kWh'))
+    if 'mtd_dollars' in discovery_info:
+        sensors.append(FplSensor(
+            hass.data[DATA_FPL], 'mtd_dollars', 'Month to Date', 'USD'))
```

Comment:
Such things should be done with the help of a dictionary lookup.

Reply:
Would you be able to go into more detail on what you mean?

Do you mean refactoring this to loop over the `discovery_info` and creating a dictionary with the timeframe and unit as values?

Reply:
List all sensors and create them if selected. See `openweathermap` or `luftdaten`.